homeassistant/components/bluetooth_le_tracker/device_tracker.py

```diff
@@ -48,10 +53,15 @@
 )
 
 
-def setup_scanner(hass, config, see, discovery_info=None):  # noqa: C901
+def setup_scanner(  # noqa: C901
+    hass: HomeAssistant,
+    config: ConfigType,
+    see: Callable[..., None],
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
     """Set up the Bluetooth LE Scanner."""
 
-    new_devices = {}
+    new_devices: dict[str, dict] = {}
```

Comment:
I'm pretty sure this type hint is correct, based on:
https://github.com/home-assistant/core/blob/bc2f4e82e3bd425918fb65381ed8c9264d1c6efc/homeassistant/components/bluetooth_le_tracker/device_tracker.py#L95